sdk/python/tests/test_offline_online_store_consistency.py

```diff
@@ -112,6 +114,79 @@ def prep_bq_fs_and_fv(
         yield fs, fv
 
 
+@contextlib.contextmanager
+def prep_redshift_fs_and_fv(
+    source_type: str,
+) -> Iterator[Tuple[FeatureStore, FeatureView]]:
+    client = aws_utils.get_redshift_data_client("us-west-2")
+    s3 = aws_utils.get_s3_resource("us-west-2")
+
+    df = create_dataset()
+
+    table_name = f"test_ingestion_{source_type}_correctness_{int(time.time())}"
+
+    offline_store = RedshiftOfflineStoreConfig(
+        cluster_id="feast-integration-tests",
+        region="us-west-2",
+        user="admin",
+        database="feast",
+        s3_staging_location="s3://feast-integration-tests/redshift/tests/ingestion",
+        iam_role="arn:aws:iam::402087665549:role/redshift_s3_access_role",
+    )
+
+    aws_utils.upload_df_to_redshift(
+        client,
+        offline_store.cluster_id,
+        offline_store.database,
+        offline_store.user,
+        s3,
+        f"{offline_store.s3_staging_location}/copy/{table_name}.parquet",
+        offline_store.iam_role,
+        table_name,
+        df,
+    )
+
+    redshift_source = RedshiftSource(
+        table=table_name if source_type == "table" else None,
+        query=f"SELECT * FROM {table_name}" if source_type == "query" else None,
+        event_timestamp_column="ts",
+        created_timestamp_column="created_ts",
+        date_partition_column="",
+        field_mapping={"ts_1": "ts", "id": "driver_id"},
+    )
+
+    fv = get_feature_view(redshift_source)
+    e = Entity(
+        name="driver",
+        description="id for driver",
+        join_key="driver_id",
+        value_type=ValueType.INT32,
+    )
+    with tempfile.TemporaryDirectory() as repo_dir_name, tempfile.TemporaryDirectory() as data_dir_name:
+        config = RepoConfig(
+            registry=str(Path(repo_dir_name) / "registry.db"),
+            project=f"test_bq_correctness_{str(uuid.uuid4()).replace('-', '')}",
+            provider="local",
+            online_store=SqliteOnlineStoreConfig(
+                path=str(Path(data_dir_name) / "online_store.db")
+            ),
+            offline_store=offline_store,
+        )
+        fs = FeatureStore(config=config)
+        fs.apply([fv, e])
+
+        yield fs, fv
+
+    # Clean up the uploaded Redshift table
+    aws_utils.execute_redshift_statement(
```

Comment:
Does this command execute even on failure?

Reply:
Yes that's how contextmanagers work (https://pymotw.com/2/contextlib/#from-generator-to-context-manager)